src/libraries/System.Net.NetworkInformation/src/System/Net/NetworkInformation/IPAddressUtil.cs

```diff
@@ -20,7 +21,11 @@ public static bool IsMulticast(IPAddress address)
             }
             else
             {
-                byte firstByte = address.GetAddressBytes()[0];
+                Span<byte> addressBytes = stackalloc byte[4];
+                address.TryWriteBytes(addressBytes, out int bytesWritten);
+                Debug.Assert(bytesWritten == 4);
+
+                byte firstByte = addressBytes[0];
```

Comment:
Can't this whole thing just be:
```suggestion
                byte firstByte = (byte)address.Address;
```
?

Reply:
(Related, @karelz, it's strange to me that IPAddress.Address remains obsoleted.  I understand we probably did so eons ago when IPv6 support was added, but to this day IPAddress.Address remains the most efficient way to extract the IPv4 address data.  I think we should reconsider the obsoletion.)

Reply:
> Can't this whole thing just be

Sure, but I just didn't want to use an obsolete function, I can change it if that's prefered